sqlglot/optimizer/simplify.py

```diff
@@ -98,17 +97,46 @@ def simplify_conjunctions(expression):
                 return right
             if right == FALSE:
                 return left
-            if left == right:
-                return left
-            if is_complement(left, right):
-                return TRUE
     elif isinstance(expression, exp.Not) and isinstance(expression.this, exp.Not):
         # double negation
         # NOT NOT x -> x
         return expression.this.this
     return expression
 
 
+def compare_and_prune(expression):
+    """
+    Sorts ANDs and ORs, removing duplicates and compliment expressions.
+    """
+    if isinstance(expression, exp.And):
+        return _compare_and_prune(expression, exp.And, FALSE, exp.and_)
+    if isinstance(expression, exp.Or):
+        return _compare_and_prune(expression, exp.Or, TRUE, exp.or_)
+    return expression
+
+
+def _compare_and_prune(expression, kind, compliment, result_func):
+    queue = [expression]
+    args = set()
+
+    while queue:
+        expression = queue.pop()
+
+        for value in expression.args.values():
+            if isinstance(value, kind):
+                queue.append(value)
+            else:
+                args.add(value)
+
+    args = sorted(args, key=lambda a: a.sql())
```

Comment:
why do we sort here? should we try to preserve order?

Reply:
no, we want a AND b == b AND a